Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -1176,9 +1176,96 @@ There are several ways to perform dynamic analysis:
 
 ### Testing enforced updating
 
+Starting from API level 21 (Android 5.0), together with the Play Core Library, apps can be forced to be updated. This mechanism is based on using the `AppUpdateManager`. Before that, other mechanisms were used, such as doing http calls to the Google Play Store, which are not as reliable as the APIs of the Play Store might change.
+Enforced updatinc can be really helpful when it comes to public key pinning (see the Testing Network communication for more details) when a pin has to be refreshed due to a certificate/public key rotation. Next, vulnerabilities are easily patched by means of forced updates.
+
+Please note that newer versions of an application will not fix security issues that are living in the back-ends to which the app communicates. Allowing an app not to communicate with it might not be enough. Having proper API-lifecycle management is key here.
+Similarly, when a user is not forced to update, do not forget to test older versions of your app against your API and/or use proper API versioning.
+
+#### Static analysis
+
+The code sample below shows the example of an app-update:
+```java
+//Part 1: check for update
+// Creates instance of the manager.
+AppUpdateManager appUpdateManager = AppUpdateManagerFactory.create(context);
+
+// Returns an intent object that you use to check for an update.
+Task<AppUpdateInfo> appUpdateInfo = appUpdateManager.getAppUpdateInfo();
+
+// Checks that the platform will allow the specified type of update.
+if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE
+      // For a flexible update, use AppUpdateType.FLEXIBLE
+      && appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.IMMEDIATE)) {
+                  // Request the update.
+
+
+                  //...Part 2: request update
+                  appUpdateManager.startUpdateFlowForResult(
+                     // Pass the intent that is returned by 'getAppUpdateInfo()'.
+                     appUpdateInfo,
+                     // Or 'AppUpdateType.FLEXIBLE' for flexible updates.
+                     AppUpdateType.IMMEDIATE,
+                     // The current activity making the update request.
+                     this,
+                     // Include a request code to later monitor this update request.
+                     MY_REQUEST_CODE);
+
+
+                     //...Part 3: check if update completed succesfully
+ @Override
+ public void onActivityResult(int requestCode, int resultCode, Intent data) {
+   if (myRequestCode == MY_REQUEST_CODE) {
+     if (myRequestCode != RESULT_OK) {
```

Comment:
myRequestCode -> resultCode

Didn't deepdive into the code, but you probably want to do something if resultCode == RESULT_OK ?

Reply:
Ha, indeed! this is an error on the google man page ^^